### ISSUE TITLE
How to make Prefab?!

### ISSUE THREAD

**Issue by smkplus** [playcanvas/engine#1500]
I couldn't make prefab and I have to do repetitious actions.please make it possible like Unity

**Comment by willeastcott** [comment]
You hate PlayCanvas? Great way to get people to work on your feature requests...

**Comment by yaustar** [comment]
The closest you can get to it at the moment is with the entity.clone() function as shown here in the documentation: https://developer.playcanvas.com/en/tutorials/procedural-levels/

**Comment by smkplus** [comment]
> The closest you can get to it at the moment is with the entity.clone() function as shown here in the documentation: https://developer.playcanvas.com/en/tutorials/procedural-levels/

I like to make prefab by dragging it and dropping in inspector\
I don't like to create prefab by code!!! I like to place it in inspector
![record_2019_02_25_19_35_11_761](https://user-images.githubusercontent.com/16706911/53350541-8725e100-3934-11e9-91f6-78ea4de2ae95.gif)

**Comment by yaustar** [comment]
Will has mentioned that it is something that is being thought about: https://forum.playcanvas.com/t/a-brand-new-upgraded-up-to-date-editor/9353/2?u=yaustar

Until then, clone() is your only option and you have to work with it. You could make a proxy entity that clones an existing entity via script quite easily. You won't be able to see the clones in the editor but can get you some of the functionality with minimal script work.

**Comment by smkplus** [comment]
> You hate PlayCanvas? Great way to get people to work on your feature requests...

sorry I was angry because I have to repeat , repeat  and repeat 
Playcanvas is very good but when it hasn't prefab feature working with it is very hard for me

**Comment by smkplus** [comment]
> Will has mentioned that it is something that is being thought about: https://forum.playcanvas.com/t/a-brand-new-upgraded-up-to-date-editor/9353/2?u=yaustar
> 
> Until then, clone() is your only option and you have to work with it. You could make a proxy entity that clones an existing entity via script quite easily. You won't be able to see the clones in the editor but can get you some of the functionality with minimal script work.

but I like to clone inside editor Unity has this feature
`https://docs.unity3d.com/ScriptReference/ExecuteInEditMode.html`

**Comment by smkplus** [comment]
> You hate PlayCanvas? Great way to get people to work on your feature requests...

willeastcott I have a question
how can I make new type in playcanvas?
I like to store my object as prefab by dragging it in Assets

![wadawd](https://user-images.githubusercontent.com/16706911/53351353-62cb0400-3936-11e9-8ddb-fdb1a48b10ec.PNG)

**Comment by yaustar** [comment]
> willeastcott I have a question
how can I make new type in playcanvas?
I like to store my object as prefab by dragging it in Assets

Prefabs like Unity don't exist in PlayCanvas so you can't do this at the moment hence the suggestions of the alternatives.

**Comment by smkplus** [comment]
> > willeastcott I have a question
> > how can I make new type in playcanvas?
> > I like to store my object as prefab by dragging it in Assets
> 
> Prefabs like Unity don't exist in PlayCanvas so you can't do this at the moment, hence the suggestions of the alternatives.

I know it I like to develop it for my self because I need it 
but I don't know how can I make it can you help me 
Just tell me what should I do to make this useful feature

**Comment by yaustar** [comment]
> I know it.I like to develop it for my self because I need it
but I don't know how can I make it can you help me
Just tell me what should I do to make this useful feature

To do something like this requires modifying the editor (which isn't open source nor part of the repo) and potentially the engine.

That said, the editor does have an undocumented and non-public API (which means that it can change at any time) that some users are exploiting as shown here: https://forum.playcanvas.com/t/released-free-aris-common-utilities/9138

TL;DR The editor is not open source so you can't change existing behaviour or add new asset types. You can write a Chrome Extension/plugin to add some functionality as shown in the thread above.

Your best bet is to create the prefab object as a JSON object and a script that references it. Then write a plugin like the user in the thread that allows you to update all entities in the scene that have that script attached.

**Comment by FaiqLaiq** [comment]
Hi everyone,
Playfab/Template system has been launched in playcanvas.

[Here](https://developer.playcanvas.com/en/user-manual/templates/) is the link for official web.
